### PR TITLE
feat(TextArea): add minHeight property, docs example

### PR DESCRIPTION
**Pull request by josie11**
Adds minHeight property to TextArea that can be used with autoHeight property to specify a minimum number of rows a TextArea should have. Referenced here: [#1647](https://github.com/Semantic-Org/Semantic-UI-React/issues/1647).

I've implemented this by setting ```this.ref.rows = minHeight```, but I'm not sure if this is the ideal way. I'd have to adjust ```assertHeight``` in the [tests](https://github.com/Semantic-Org/Semantic-UI-React/blob/master/test/specs/addons/TextArea/TextArea-test.js#L68) in order to add a test for it.

### CODE REVIEW

File: src/addons/TextArea/TextArea.js [hunk 1]
```diff
@@ -75,14 +79,14 @@ class TextArea extends Component {
...
-    const { autoHeight } = this.props
+    const { autoHeight, minHeight } = this.props
     if (!autoHeight) return
 
     let { borderTopWidth, borderBottomWidth } = window.getComputedStyle(this.ref)
     borderTopWidth = parseInt(borderTopWidth, 10)
     borderBottomWidth = parseInt(borderBottomWidth, 10)
 
-    this.ref.rows = '1'
+    this.ref.rows = minHeight
```
Comment (layershifter) [inline src/addons/TextArea/TextArea.js]: `minHeight` prop should affect next line,  `rows` should be also customizable:

```diff
static defaultProps = {
   as: 'textarea',
+  minHeight: 0,
+  rows: 5,
}
```

```diff
-this.ref.rows = minHeight
-this.ref.style.minHeight = '0'
+this.ref.style.minHeight = minHeight + 'px'
```

```diff
render() {
-  const { value } = this.props
+  const { rows, value } = this.props
// ...
- return <ElementType{...rest} onChange={this.handleChange} ref={this.handleRef} value={value} />
+  return (
+    <ElementType
+      {...rest} 
+      onChange={this.handleChange}
+      ref={this.handleRef}
+      rows={rows}
+      value={value}
+    />
+  )
}
```

**Review by layershifter — CHANGES_REQUESTED** [review]
Thanks for PR. I think that `rows` should be also customizable:

```diff
static defaultProps = {
   as: 'textarea',
+  minHeight: 0,
+  rows: 5,
}
```

```diff
-this.ref.rows = minHeight
-this.ref.style.minHeight = '0'
+this.ref.style.minHeight = minHeight + 'px'
```

```diff
render() {
-  const { value } = this.props
+  const { rows, value } = this.props
// ...
- return <ElementType{...rest} onChange={this.handleChange} ref={this.handleRef} value={value} />
+  return (
+    <ElementType
+      {...rest} 
+      onChange={this.handleChange}
+      ref={this.handleRef}
+      rows={rows}
+      value={value}
+    />
+  )
}
```

**Comment by josie11** [comment]
I added the ```rows``` prop but I left it in ```updateHeight``` & usable with ```autoHeight``` for now - otherwise it will break tests. Also, when using autoHeight it seems the intended default rows should be 1, so there is a conflict there.

**Review by layershifter — APPROVED** [review]
I've pushed some changes there, LGTM.

However, there is one breaking change with `autoHeight`. Before these changes `rows` prop had `1` value, now it's has `3` by default.

**Comment by levithomason** [comment]
Reviewing the code changes, it strikes me odd that we're pulling `autoHeight` up to a prop.  Perhaps it makes more sense to leave this in the style prop:

```jsx
<TextArea style={{ minHeight: 100 }} />
```

**Comment by layershifter** [comment]
@levithomason Thanks for the feedback, I'll make these changes.

**Review by layershifter — APPROVED** [review]
I've performed there some changes. `minHeight` was back to style :+1:

**Review by levithomason — APPROVED** [review]
Thanks much @josie11 and @layershifter!

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.69.0`

**Pull request merged**